src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2809,10 +2809,16 @@ MavlinkReceiver::Run()
 			updateParams();
 		}
 
-		if (poll(&fds[0], 1, timeout) > 0) {
+		int ret = poll(&fds[0], 1, timeout);
+
+		if (ret > 0) {
 			if (_mavlink->get_protocol() == Protocol::SERIAL) {
 				/* non-blocking read. read may return negative values */
 				nread = ::read(fds[0].fd, buf, sizeof(buf));
+
+				if (nread == -1 && errno == ENOTCONN) { // Not connected (can happen for USB)
+					usleep(100000);
```

Comment:
Detail: time literal could be used

Reply:
We should only use our time literals in the context of hrt. The base unit for hrt is currently microseconds, so coincidentally it currently fits in usleep, but it's possible it could change someday.

Reply:
I see, always thought usleep and hrt and everything not float is microseconds in PX4 and hence we can use the literals. But that seems to be a naive assumption. 👍